src/UnderlineNav2/UnderlineNavItem.tsx

```diff
@@ -70,30 +77,51 @@ export const UnderlineNavLink = forwardRef(
     const textStyles: BetterSystemStyleObject = {
       whiteSpace: 'nowrap'
     }
+
+    const wrapperStyles = {
```

Comment:
I added another Box (div) that wraps icon, text and the counter and gave some padding to achieve smaller grey rectangular hover state.
<img width="181" alt="Screen Shot 2022-08-29 at 3 12 39 pm" src="https://user-images.githubusercontent.com/1446503/187127575-950233ea-ae0d-42d3-8596-1f361e086b1c.png">